retrieval_service/datastore/providers/firestore.py

```diff
@@ -42,41 +45,121 @@ def kind(cls):
 
     def __init__(self, client: AsyncClient):
         self.__client = client
+        self.__policies_collection = AsyncQuery(self.__client.collection("policies"))
+        self.__amenities_collection = AsyncQuery(self.__client.collection("amenities"))
 
     @classmethod
     async def create(cls, config: Config) -> "Client":
         return cls(AsyncClient(project=config.projectId))
 
+    async def __delete_collections(
+        self, collection_list: list[AsyncCollectionReference]
+    ):
+        # Checks if collection exists and deletes all documents
+        delete_tasks = []
+        for collection_ref in collection_list:
+            collection_exists = collection_ref.limit(1).stream()
+            if not collection_exists:
+                continue
+
+            docs = collection_ref.stream()
+            async for doc in docs:
+                delete_tasks.append(asyncio.create_task(doc.reference.delete()))
+        await asyncio.gather(*delete_tasks)
+
+    async def __list_vector_index(self) -> list[str]:
+        list_vector_index_command = [
+            "gcloud",
+            "alpha",
+            "firestore",
+            "indexes",
+            "composite",
+            "list",
+            "--database=(default)",
+            "--format=value(name)",  # prints name field
+        ]
+
+        list_vector_index_process = await asyncio.create_subprocess_exec(
+            *list_vector_index_command,
```

Comment:
nit: consider if it's really useful to put this in it's own array as opposed to just inlining in the command.